### ISSUE TITLE
Scan stream does not "end" when the set does not exist

### ISSUE THREAD

**Issue by revington** [aerospike/aerospike-client-nodejs#132]
A scan will never `end` if we try to scan an aerospike cluster with no data.
The js code:
```
const AEROSPIKE_NS = 'myns';   
const AEROSPIKE_SET = 'myset';            

 Aerospike.connect(aerospikeConfig, function (err, aerospikeClient) {        
     if (err) {                                                              
         console.error(now(), 'Aerospike connection error', err);            
         console.trace(err.stack);                                           
         return process.exit(1);                                             
     }                                                                       
     let scan = aerospikeClient.scan(AEROSPIKE_NS, AEROSPIKE_SET);           
     scan.priority = Aerospike.scanPriority.LOW;                             
     scan.percent = 100;                                                     
     scan.nobins = false;                                                    
     let scanStream = scan.foreach();                                        
     scanStream.on('error', function (err) {                                 
         console.error(now(), 'AEROSPIKE SCAN ERR', err);                    
         console.trace(err.stack);                                           
         errors++;                                                           
     });                                                                     
                                                                             
     scanStream.on('end', function () {                                                                                   
         flush();                                                            
         console.error('records read from aerospike', recordsRead, 'errors', errors);
         aerospikeClient.close();                                            
     });                                                                     
                                                                             
     scanStream.on('data', function (record) {                               
         recordsRead++;                                                      
         recordWriter.push(record);                                          
         if (recordWriter.size() > BUFFER_SIZE) {                            
             flush();                                                        
         }                                                                   
     });                                                                     
 });                                                                         
```

Here is the log 
```
TRACE(7690) [conversions.cc:84] [config_from_jsobject] - host[0].addr = "172.17.0.2"
TRACE(7690) [conversions.cc:93] [config_from_jsobject] - host[0].port = 3000
DEBUG(7690) [conversions.cc:248] [config_from_jsobject] - Could not find a valid LUA system path ./aerospike-client-c/package/opt/aerospike/client/sys/udf/lua/
DEBUG(7690) [conversions.cc:273] [config_from_jsobject] - Could not find valid LUA user path ./aerospike-client-c/package/opt/aerospike/client/usr/udf/lua
DEBUG(7690) [client.cc:115] [New] - Aerospike object initialization : success
DEBUG(7690) [cluster.cc:52] [Connect] - Connecting to Cluster: Success
INFO (7690) [scan_async.cc:71] [ScanAsync] - Using scan ID 1209261024100961 for async scan.
DEBUG(7690) [scan_async.cc:74] [ScanAsync] - Sending async scan command
```

**Comment by jhecking** [comment]
@revington, agree, this is not the expected behavior. I believe the issue is with the Aerospike C client library and I am working with the maintainer on a fix.

**Comment by jhecking** [comment]
Resolved in v2.1.1 by upgrading to C client v4.0.6.

**Comment by revington** [comment]
thx!
